### PR TITLE
feat!: replace tslog by awspowertools logging

### PR DESCRIPTION
**Pull request by npalm**
The module is using TSLog now for a sometime. But after breaking changes we are not able to update TSLog anymore. In the meantime, AWS have released [AWS Powertools for Lambda](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/), including logging capabilities. Since this solution is AWS native, we are replacing the outdated TSLog by AWS Powertools logging.

## Breaking change
In this PR we replace TSLog by AWS Powertools logs with a minimal impact on code changes. The change can have be breaking in the following cases:
- Processing the logging, we only will support JSON formatted log
- Depending on the removed setting LOG_TYPE 
- Depending on the LOG_LEVEL silly, fatal or trace

## Log messages
All log messages are JSON formatted and contain besides the message scope information like: `function-name`, `environment`, `module`. By enabling DEBUG log level the event received by the Lambda will be logged.

### CODE REVIEW

**Comment by npalm** [comment]
Example log message:

```json
{
    "level": "INFO",
    "message": "Created instance(s): i-0532e57fcbb2c5276",
    "service": "runners-scale-up",
    "timestamp": "2023-03-15T10:28:31.368Z",
    "xray_trace_id": "1-64119dc9-2ecba0ebb133eb0c9bb77a0a",
    "module": "runners",
    "region": "eu-west-1",
    "environment": "np-default",
    "aws-request-id": "61c8abbd-5398-506f-b748-f736cc61a60d",
    "function-name": "np-default-scale-up",
    "data": {
        "runnerType": "Org",
        "runnerOwner": "test-runners",
        "event": "workflow_job",
        "id": "12015854576"
    }
}
```

File: modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/package.json [hunk 1]
```diff
@@ -35,8 +36,8 @@
     "typescript": "^4.9.5"
   },
   "dependencies": {
-    "axios": "^1.3.3",
+    "@aws-lambda-powertools/logger": "^1.6.0",
```
[the remark this answers is not in the document]
Comment (npalm) [reply modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/package.json:39]: what do you mean? Add the reference where?

**Review by GuptaNavdeep1983 — APPROVED** [review]

File: modules/runners/pool/main.tf [hunk 2]
```diff
@@ -17,6 +17,7 @@ resource "aws_lambda_function" "pool" {
 
   environment {
     variables = {
+      AMI_ID_SSM_PARAMETER_NAME            = var.config.ami_id_ssm_parameter_name
```
Comment (GuptaNavdeep1983) [inline modules/runners/pool/main.tf:20]: why is this change here, is it not already addressed as part of another PR?

File: modules/runners/lambdas/runners/src/scale-runners/scale-up.ts [hunk 3]
```diff
@@ -171,13 +165,18 @@ export async function scaleUp(eventSource: string, payload: ActionRequestMessage
   const runnerType = enableOrgLevel ? 'Org' : 'Repo';
   const runnerOwner = enableOrgLevel ? payload.repositoryOwner : `${payload.repositoryOwner}/${payload.repositoryName}`;
 
-  LogFields.fields = {};
-  LogFields.fields.runnerType = runnerType;
-  LogFields.fields.runnerOwner = runnerOwner;
-  LogFields.fields.event = payload.eventType;
-  LogFields.fields.id = payload.id.toString();
+  addPersistentContextToChildLogger({
```
[the remark this answers is not in the document]
Comment (npalm) [reply modules/runners/lambdas/runners/src/scale-runners/scale-up.ts:168]: this ensure all log statements written by any child have the same persistent context

**Comment by GuptaNavdeep1983** [comment]
Tested the changes in this PR using default example and the logging changes are reflected in cloud watch log group.

**Review by GuptaNavdeep1983 — DISMISSED** [review]

**Review by ScottGuymer — APPROVED** [review]
